### PR TITLE
updated error handling for windows ofi_shm_map

### PR DESCRIPTION
**Pull request by hoopoepg**
updated error handling for windows implementation
of ofi_shm_map function to follow common code styling

Signed-off-by: Oblomov, Sergey <sergey.oblomov@intel.com>

### CODE REVIEW

**Comment by lanl-ompi** [comment]
bot:retest

File: src/windows/osd.c [hunk 1]
```diff
@@ -207,33 +207,36 @@ int ofi_shm_map(struct util_shm *shm, const char *name, size_t size,
...
 
-fn_fail:
-	if (shm->shared_fd)
-		CloseHandle(shm->shared_fd);
-	if (fname)
-		free((void*)fname);
+fn_nomap:
+	CloseHandle(shm->shared_fd);
+fn_nofilemap:
+	free((void*)fname);
```
Comment (hoopoepg) [inline src/windows/osd.c]: yes, but windows compiler prompts warning without casting

**Comment by goodell** [comment]
other than a minor nit, the change is good to me 👍 

I randomly noticed that your author/commit names are inconsistent:

```
$ git log --author=oblomov --author=Sergey --author=hoopoepg --committer=oblomov --committer=Sergey --committer=hoopoepg --pretty=tformat:$'%aN <%aE>\n%cN <%cE>' | sort | uniq -c | sort -r -n
     17 soblomov <sergey.oblomov@intel.com>
      4 Oblomov, Sergey <sergey.oblomov@intel.com>
      3 Sergey Oblomov <sergey.oblomov@intel.com>
      2 Oblomov, Sergey <hoopoepg@gmail.com>
```

You might consider making sure your `~/.gitconfig` `user.name`/`user.email` settings are consistent across your development machines.  You may also want to file a PR to change the libfabric `.mailmap` file to unify your name/email for the existing mix of commits.

File: src/windows/osd.c (hunk 1 above)
Comment (goodell) [inline src/windows/osd.c]: bizarre, does it compile in C++ mode by default or something?

**Comment by hoopoepg** [comment]
yep, my fault. some of commits are from windows system, other from linux, some from my home laptop (when I'm working at night :) ) there are different settings. I will sync commit signatures.
thank you

File: src/windows/osd.c (hunk 1 above)
Comment (hoopoepg) [inline src/windows/osd.c]: nop, it is just paranioc compiler configuration :)

Comment (jsquyres) [inline src/windows/osd.c]: FWIW: I'm all for paranoid compiler configurations, but one that forces excessive unnecessary casts is not a good configuration.

Comment (hoopoepg) [inline src/windows/osd.c]: fixed

**Pull request merged**